src/app/services/dsm.service.ts

```diff
@@ -156,16 +183,20 @@ export class DSMService {
     map.push( {name: "to", value: to} );
     map.push( {name: "userId", value: userId} );
     map.push( {name: "parent", value: parent} );
+
     if (filterQuery != null) {
       map.push( {name: "filterQuery", value: filterQuery} );
     }
-    else if (json == null || json.filters == undefined || json.filters == null) {
-      map.push( {name: "filterName", value: json == null ? null : json.filterName} );
+    else if (json == null || json.filters == null) {
+      json && map.push( {name: "filterName", value: json.filterName} );
```

Comment:
Is that `json &&` supposed to be here?

Reply:
I refactored that in purpose as to avoid `filterName=null` in query params (DSM server API sent an error in this case).

`json && map.push( {name: "filterName", value: json.filterName} );` means that we should set `filterName` param 
if `json` is **not**  `null` only.
(I suspect, it is a case when there are not any search/custom filters, but there is a default one.)